src/Widgets/VPN/VPNInfoDialog.vala

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (c) 2015-2019 elementary, Inc. (https://elementary.io)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Authored by: Adam Bieńkowski <donadigos159@gmail.com>
+ */
+
+public class Network.Widgets.VPNInfoDialog : Gtk.Dialog {
+    public NM.RemoteConnection? connection { get; construct; }
+    private string service_type;
+
+    private Gtk.Label vpn_type;
+    private Gtk.Label gateway;
+    private Gtk.Label username;
+
+    public string state { get; construct; }
+
+    public VPNInfoDialog (string state, NM.RemoteConnection connection) {
+        Object (
+            deletable: false,
+            modal: true,
+            resizable: true,
+            state: state,
+            connection: connection
+        );
+    }
+
+    construct {
+        var image = new Gtk.Image.from_icon_name ("network-vpn", Gtk.IconSize.DIALOG);
+        image.hexpand = true;
+        image.halign = Gtk.Align.CENTER;
+
+        var data_grid = new Gtk.Grid ();
+        data_grid.expand = true;
+        data_grid.halign = Gtk.Align.START;
+
+        var state_label = new Gtk.Label (state);
+        state_label.selectable = true;
+        state_label.xalign = 0;
+
+        vpn_type = new Gtk.Label ("");
+        vpn_type.selectable = true;
+        vpn_type.xalign = 0;
+        vpn_type.no_show_all = true;
+
+        username = new Gtk.Label ("");
+        username.selectable = true;
+        username.xalign = 0;
+        username.no_show_all = true;
+
+        gateway = new Gtk.Label ("");
+        gateway.selectable = true;
+        gateway.xalign = 0;
+        gateway.no_show_all = true;
+
+        data_grid.attach (new VPNInfoLabel (_("Status: ")), 0, 0, 1, 1);
```

Comment:
Never end a string with a space, use the column spacing property of a GtkGrid to do that
```suggestion
        data_grid.attach (new VPNInfoLabel (_("Status:")), 0, 0, 1, 1);
```